src/plugins/data/public/ui/dataset_selector/dataset_selector.tsx

```diff
@@ -82,7 +82,8 @@ export const DatasetSelector = ({
   const { overlays } = services;
   const datasetService = getQueryService().queryString.getDatasetService();
   const datasetIcon =
-    datasetService.getType(selectedDataset?.type || '')?.meta.icon.type || 'database';
+    datasetService.getType(selectedDataset?.ref?.type || selectedDataset?.type || '')?.meta.icon
```

Comment:
q. is indexView the only use case for `ref` right now? do we have plan to add other use cases?

Reply:
For now it is only for index views